collector/mongod/replset_status.go

```diff
@@ -1,232 +1,223 @@
 package collector_mongod
 
 import (
-    "time"
-    "github.com/golang/glog"
-    "github.com/prometheus/client_golang/prometheus"
-    "gopkg.in/mgo.v2"
-    "gopkg.in/mgo.v2/bson"
+	"time"
+
+	"gopkg.in/mgo.v2"
+	"gopkg.in/mgo.v2/bson"
+
+	"github.com/golang/glog"
+	"github.com/prometheus/client_golang/prometheus"
 )
 
 var (
-    replSetLastElection = prometheus.NewCounterVec(prometheus.CounterOpts{
-            Namespace: Namespace,
-            Name:      "replset_last",
-            Help:      "Last event times for replica set",
-    }, []string{"event"})
-    replSetTotalMembers = prometheus.NewGauge(prometheus.GaugeOpts{
-            Namespace: Namespace,
-            Subsystem: "replset",
-            Name:      "members",
-            Help:      "Number of members in replica set",
-    })
-    replSetTotalMembersWithData = prometheus.NewGauge(prometheus.GaugeOpts{
-            Namespace: Namespace,
-            Subsystem: "replset",
-            Name:      "members_w_data",
-            Help:      "Number of members in replica set with data",
-    })
-    replSetMyLagMs = prometheus.NewGauge(prometheus.GaugeOpts{
-            Namespace: Namespace,
-            Subsystem: "replset",
-            Name:      "my_lag_ms",
-            Help:      "Lag in milliseconds in reference to replica set Primary node",
-    })
-    replSetMaxNode2NodePingMs = prometheus.NewGauge(prometheus.GaugeOpts{
-            Namespace: Namespace,
-            Subsystem: "replset",
-            Name:      "max_n2n_ping_ms",
-            Help:      "Maximum ping in milliseconds to other replica set members",
-    })
+	myState   = prometheus.NewGaugeVec(prometheus.GaugeOpts{
+		Namespace: Namespace,
+		Subsystem: "replset",
+		Name:      "my_state",
+		Help:      "An integer between 0 and 10 that represents the replica state of the current member",
+	}, []string{"set"})
+	term = prometheus.NewGaugeVec(prometheus.GaugeOpts{
+		Namespace: Namespace,
+		Subsystem: "replset",
+		Name:      "term",
+		Help:      "The election count for the replica set, as known to this replica set member",
+	}, []string{"set"})
+	numberOfMembers = prometheus.NewGaugeVec(prometheus.GaugeOpts{
+		Namespace: Namespace,
+		Subsystem: "replset",
+		Name:      "number_of_members",
+		Help:      "The number of replica set mebers",
+	}, []string{"set"})
+	heartbeatIntervalMillis = prometheus.NewGaugeVec(prometheus.GaugeOpts{
+		Namespace: Namespace,
+		Subsystem: "replset",
+		Name:      "heatbeat_interval_millis",
+		Help:      "The frequency in milliseconds of the heartbeats",
+	}, []string{"set"})
+	memberHealth = prometheus.NewGaugeVec(prometheus.GaugeOpts{
+		Namespace: Namespace,
+		Subsystem: "replset",
+		Name:      "member_health",
+		Help:      "This field conveys if the member is up (1) or down (0).",
+	}, []string{"set", "name", "state"})
```

Comment:
replset_status.go is an improved version of my code pulled down from mainline, so this was a choice from the upstream committer. There is no extra work being done to gather these metrics seeing they're already in the result we require, but we can disable anything you think is extraneous if they're not used in graphs and space is a concern